sabnzbd/downloader.py

```diff
@@ -861,25 +829,68 @@ def __handle_recv_result(self, nw: NewsWrapper, bytes_received: int = 0, done: b
             server.bad_cons = 0
             server.errormsg = server.warning = ""
 
-            # Update statistics and decode
-            article.nzf.nzo.update_download_stats(sabnzbd.BPSMeter.bps, server.id, nw.data_position)
-            self.decode(article, nw.get_data_buffer(), nw.data_position)
+            # Update statistics
+            with sabnzbd.decorators.NZBQUEUE_LOCK:
+                article.nzf.nzo.update_download_stats(sabnzbd.BPSMeter.bps, server.id, nw.data_position)
```

Comment:
Do you think this can move this above the status_code check where a lock is already acquired, with an `if done:`

`done` is true if `\r\n.\r\n` is found which I think is only at the end of of an article?

Reply:
Yes I think so.